great_expectations/experimental/datasources/data_asset/data_connector/filesystem_data_connector.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import logging
+import pathlib
+import re
+from typing import List
+
+from great_expectations.datasource.data_connector.util import (
+    get_filesystem_one_level_directory_glob_path_list,
+    normalize_directory_path,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector import (
+    FilePathDataConnector,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class FilesystemDataConnector(FilePathDataConnector):
```

Comment:
many interface methods can be removed, data_asset_name can likely be removed

Reply:
@donaldheppner These interface methods are important for subclasses to implement; otherwise, we will not be able to guarantee performance and certain prescribed functionality.  Furthermore, just like `data_asset_name` is currently supplied to create `BatchDefinition` in the lowest leaf filesystem data asset in ZEP today, it is needed for `DataConnector`, now that creation and manipulation of `BatchDefinition` is being migrated upwards.